### PR TITLE
feat: format umd

### PR DESCRIPTION
**Pull request by underfin**

### CODE REVIEW

File: crates/rolldown/tests/rolldown/function/export_mode/umd/complex_name_named/_config.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "config": {
+    "format": "umd",
+    "name": "test.module",
+    "exports": "named",
+    "banner": "import assert from \"node:assert\"",
+    "footer": "assert(globalThis.test.module.a === 1);"
```
Comment (hyf0) [inline crates/rolldown/tests/rolldown/function/export_mode/umd/complex_name_named/_config.json]: How about using `_test.mjs`? This seems to abuse the banner and footer.

```js
// __test.mjs
import assert from "node:assert"
import "./dist/main.js"
assert(globalThis.test.module.a === 1);
```

File: crates/rolldown/tests/rolldown/function/export_mode/umd/complex_name_named/artifacts.snap [hunk 2]
```diff
@@ -0,0 +1,24 @@
+---
+source: crates/rolldown_testing/src/integration_test.rs
+---
+# Assets
+
+## main.js
+
+```js
+import assert from "node:assert"
```
Comment (hyf0) [inline crates/rolldown/tests/rolldown/function/export_mode/umd/complex_name_named/artifacts.snap]: Would be better if this doesn't show in the snapshot. https://github.com/rolldown/rolldown/pull/2489/files#r1812325035

File: crates/rolldown/tests/rolldown/function/export_mode/umd/complex_name_named/_config.json (hunk 1 above)
Comment (underfin) [reply crates/rolldown/tests/rolldown/function/export_mode/umd/complex_name_named/_config.json]: Good idea,Fixed it.

**Review by hyf0 — APPROVED** [review]
Good job!

**Pull request merged**